soh/soh/OTRGlobals.cpp

```diff
@@ -1966,24 +1968,24 @@ extern "C" int CustomMessage_RetrieveIfExists(PlayState* play) {
             }
 
             RandomizerCheck hintCheck =
-                Randomizer_GetCheckFromActor(msgCtx->talkActor->id, play->sceneNum, actorParams);
+                Randomizer_GetCheckFromActor(stone->id, play->sceneNum, actorParams);
 
             messageEntry = CustomMessageManager::Instance->RetrieveMessage(Randomizer::hintMessageTableID, hintCheck);
-        } else if (textId == TEXT_ALTAR_CHILD || textId == TEXT_ALTAR_ADULT) {
+        } else if ((textId == TEXT_ALTAR_CHILD || textId == TEXT_ALTAR_ADULT) && Randomizer_GetSettingValue(RSK_TOT_ALTAR_HINT)) {
             // rando hints at altar
             messageEntry = (LINK_IS_ADULT)
                ? CustomMessageManager::Instance->RetrieveMessage(Randomizer::hintMessageTableID, TEXT_ALTAR_ADULT)
                : CustomMessageManager::Instance->RetrieveMessage(Randomizer::hintMessageTableID, TEXT_ALTAR_CHILD);
         } else if (textId == TEXT_GANONDORF) {
-            if (INV_CONTENT(ITEM_ARROW_LIGHT) == ITEM_ARROW_LIGHT) {
+            if (INV_CONTENT(ITEM_ARROW_LIGHT) == ITEM_ARROW_LIGHT || !Randomizer_GetSettingValue(RSK_GANONDORF_LIGHT_ARROWS_HINT)) {
```

Comment:
Depending on how we want to handle the no-hint ganon text, maybe this RSK check needs to go one line higher with `text == TEXT_GANONDORF` so the original ganon text can play.

Reply:
I'll let OTRGlobals handle the text for Ganondorf since the ganon hint will always be generated.